x-pack/plugins/lens/public/indexpattern_datasource/datapanel.tsx

```diff
@@ -443,98 +488,170 @@ export const InnerIndexPatternDataPanel = function InnerIndexPatternDataPanel({
+                          data={data}
+                          indexPattern={currentIndexPattern}
+                          key={field.name}
+                          field={field}
+                          highlight={hilight}
+                          exists={false}
+                          dateRange={dateRange}
+                          query={query}
+                          filters={filters}
+                        />
+                      );
+                    })}
+                  </div>
+                ) : (
+                  <NoFieldsCallout
+                    isAffectedByFieldFilter={hasFieldFilter}
+                    existFieldsInIndex={!!fieldGroups.emptyFields.length}
                   />
-                );
-              })}
-
-              {paginatedFields.length === 0 && (
-                <EuiCallOut
-                  size="s"
-                  color="warning"
-                  title={
-                    localState.typeFilter.length || localState.nameFilter.length
-                      ? i18n.translate('xpack.lens.indexPatterns.noFilteredFieldsLabel', {
-                          defaultMessage: 'No fields match the current filters.',
-                        })
-                      : showEmptyFields
-                      ? i18n.translate('xpack.lens.indexPatterns.noFieldsLabel', {
-                          defaultMessage: 'No fields exist in this index pattern.',
-                        })
-                      : i18n.translate('xpack.lens.indexPatterns.emptyFieldsWithDataLabel', {
-                          defaultMessage: 'Looks like you don’t have any data.',
-                        })
-                  }
-                >
-                  {(!showEmptyFields ||
-                    localState.typeFilter.length ||
-                    localState.nameFilter.length) && (
-                    <>
-                      <strong>
-                        {i18n.translate('xpack.lens.indexPatterns.noFields.tryText', {
-                          defaultMessage: 'Try:',
-                        })}
-                      </strong>
-                      <ul>
-                        <li>
-                          {i18n.translate('xpack.lens.indexPatterns.noFields.extendTimeBullet', {
-                            defaultMessage: 'Extending the time range',
-                          })}
-                        </li>
-                        <li>
-                          {i18n.translate('xpack.lens.indexPatterns.noFields.fieldFilterBullet', {
-                            defaultMessage:
-                              'Using {filterByTypeLabel} {arrow} to show fields without data',
-                            values: { filterByTypeLabel, arrow: '↑' },
-                          })}
-                        </li>
-                      </ul>
-                    </>
-                  )}
-                </EuiCallOut>
-              )}
+                )}
+              </EuiAccordion>
```

Comment:
Both accordions seem to share a lot of common code: it might not be worth it, but can these be made into a function?